main.tf

```diff
@@ -0,0 +1,108 @@
+locals {
+  name = "${var.name}-${random_string.this.result}"
+}
+
+# Creating a random string for IAM Role name interpolation
+resource "random_string" "this" {
+  length  = 5
+  special = false
+}
+
+data "aws_iam_policy_document" "assume" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type = "Service"
+
+      identifiers = [
+        "ec2.amazonaws.com",
+        "ssm.amazonaws.com",
+      ]
+    }
+
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+resource "aws_iam_role" "this" {
+  name               = local.name
+  assume_role_policy = data.aws_iam_policy_document.assume.json
+}
+
+resource "aws_iam_role_policy_attachment" "this" {
+  for_each   = toset(["arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore", "arn:aws:iam::aws:policy/CloudWatchAgentServerPolicy"])
+  role       = aws_iam_role.this.name
+  policy_arn = each.key
+}
+
+resource "aws_iam_instance_profile" "this" {
+  name = local.name
+  role = aws_iam_role.this.name
+}
+
+data "aws_ami" "amazon_linux_2" {
+  most_recent = true
+
+
+  filter {
+    name   = "owner-alias"
+    values = ["amazon"]
+  }
+
+  owners = ["amazon"] # Canonical
+
+  filter {
+    name   = "name"
+    values = ["amzn2-ami-hvm*"]
+  }
+}
+
+
+resource "aws_security_group" "allow_egress" {
+  count       = var.create_security_group ? 1 : 0
+  name        = local.name
+  description = "Allow egress traffic for ${local.name}"
+  vpc_id      = var.vpc_id
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+data "template_file" "init" {
+  template = file("${path.module}/cloud_init.init")
+}
+
+## Creating Launch Configuration
+resource "aws_launch_configuration" "this" {
+  image_id                    = data.aws_ami.amazon_linux_2.id
+  instance_type               = "t2.micro"
```

Comment:
`t3.nano` would not be enough?

Reply:
It can be, it's a bit tight though when many people are on there. It should be adjustable.